preup/kickstart/application.py

```diff
@@ -210,10 +210,28 @@ def comment_kickstart_issues(self):
                 kickstart_data[index] = "#" + row
         FileHelper.write_to_file(self.kick_start_name, 'wb', kickstart_data)
 
+    def _get_all_postinstall_scripts(self):
+        found_scripts = []
+        for (path, dirs, files) in os.walk(os.path.join(settings.result_dir, settings.kickstart_dir)):
+            for f in files:
+                file_name = os.path.join(path, f)
+                if os.path.exists(file_name) and FileHelper.check_file(file_name, u"x") is True:
+                    found_scripts.append(file_name)
+        return found_scripts
+
     def generate(self):
         if not self.collect_data():
             log_message("Important data are missing for the Kickstart generation.", level=logging.ERROR)
             return None
+        if not self._get_all_postinstall_scripts():
```

Comment:
The implementation of this whole if condition is quite low level (level of abstraction). However the method generate is very high level. We should not mix these different levels in one function/method. Please move this condition into a separate method. Then just one line like the following would be here: "